features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt

```diff
@@ -334,6 +335,49 @@ interface RumMonitor {
     @ExperimentalRumApi
     fun addViewLoadingTime(overwrite: Boolean)
 
+    /**
+     * Starts the [name] feature operation.
+     *
+     * @param name the name of the operation.
+     * @param operationKey optional operation key. Allows to track multiple operations of the same [name].
+     * For example, multiple network requests (photo or file uploads) for the same URL.
+     * @param attributes additional custom attributes to attach to the feature operation.
+     */
+    @ExperimentalRumApi
+    fun startFeatureOperation(name: String, operationKey: String? = null, attributes: Map<String, Any?> = emptyMap())
+
+    /**
+     * Finishes the [name] feature operation with successful status.
+     *
+     * @param name the name of the operation.
+     * @param operationKey optional operation key identifying a specific operation
+     * instance from the list of feature operations of the same [name]. Should be provided if [operationKey] was
+     * provided during [startFeatureOperation] invocation.
+     * @param attributes additional custom attributes to attach to the feature operation. Can be
+     * used to add some result data produced as the result of the operation.
+     */
+    @ExperimentalRumApi
+    fun succeedFeatureOperation(name: String, operationKey: String? = null, attributes: Map<String, Any?> = emptyMap())
+
+    /**
+     * Finishes the [name] feature operation with failure status.
+     *
+     * @param name the name of the operation.
+     * @param operationKey optional operation key identifying a specific operation
+     * instance from the list of feature operations of the same [name]. Should be provided if [operationKey] was
+     * provided during [startFeatureOperation] invocation.
+     * @param failureReason the reason for the operation failure.
+     * @param attributes additional custom attributes to attach to the feature operation. Can be
+     * used to add some result data produced as the result of the operation.
+     */
+    @ExperimentalRumApi
+    fun failFeatureOperation(
+        name: String,
+        operationKey: String? = null,
+        failureReason: FailureReason,
```

Comment:
FYI: on iOS this attribute is called just `reason`, not `failureReason`, so it will be nice to align

https://github.com/DataDog/dd-sdk-ios/pull/2427/files#diff-8a79dd186c7b2bd17b0d53e57ebddbfd05909f395af9a6d01e9ac6c46b9a1f22R341